src/volume_mgr.rs

```diff
@@ -103,7 +103,36 @@ where
         &mut self,
         volume_idx: VolumeIdx,
     ) -> Result<Volume<D, T, MAX_DIRS, MAX_FILES, MAX_VOLUMES>, Error<D::Error>> {
-        let v = self.open_raw_volume(volume_idx)?;
+        return self._open_volume(volume_idx, false);
+    }
+
+    /// Get a read only volume (or partition) based on entries in the Master Boot Record.
+    /// Opening and closing a read only volume is faster than a writable volume.
+    ///
+    /// We do not support GUID Partition Table disks. Nor do we support any
+    /// concept of drive letters - that is for a higher layer to handle.
+    pub fn open_volume_read_only(
+        &mut self,
+        volume_idx: VolumeIdx,
+    ) -> Result<Volume<D, T, MAX_DIRS, MAX_FILES, MAX_VOLUMES>, Error<D::Error>> {
+        return self._open_volume(volume_idx, true);
+    }
+
+    /// Get a volume (or partition) based on entries in the Master Boot Record.
+    ///
+    /// We do not support GUID Partition Table disks. Nor do we support any
+    /// concept of drive letters - that is for a higher layer to handle.
+    fn _open_volume(
```

Comment:
We added an argument to `open_raw_volume`, so should we just add an argument to `open_volume`, and not have the two extra methods? Not sure.

Reply:
I made it like this, so it is downward compatible

Reply:
Then should we keep open_raw_volume backwards compatible too?

Reply:
Is there a need for? I though because it will be used mostly internally so the impact is lower?

Reply:
it's a public API, which I use in my projects.

Reply:
I can change also to the same structure as for open_volume or do you prefer the argument way for both?

Reply:
I think I prefer it with the argument.

Reply:
Perfect then I will change it also for the open_volume

Reply:
fixed in 70a5fdbe7d9aa69273baa77fc23012ea9c34666d